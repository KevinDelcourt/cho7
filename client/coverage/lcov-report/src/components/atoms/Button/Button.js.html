<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/atoms/Button/Button.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">src/components/atoms/Button</a> Button.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from "styled-components"
&nbsp;
const Button = styled.button`
    background-color: ${props =&gt; props.bgColor};
    border: ${props =&gt; props.border || "none"};
    border-radius: 5px;
    padding: 5px 15px;
    height: 35px;
    font-size:18px;
    color:${props =&gt; props.color || "black"}
`;
&nbsp;
export default Button
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Apr 03 2019 11:16:41 GMT+0200 (GMT+02:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
