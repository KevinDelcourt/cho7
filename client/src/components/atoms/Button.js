import styled from "styled-components"

const Button = styled.button`
    background-color: ${props => props.bgColor};
    border: ${props => props.border || "none"};
    border-radius: 5px;
    padding: 5px 15px;
    height: 35px;
    font-size: 18px;
`

export default Button
