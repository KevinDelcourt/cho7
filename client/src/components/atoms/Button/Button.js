import styled from "styled-components"
import { connect } from "react-redux"

const Button = styled.button`
    background-color: ${props => props.bgColor};
    border: ${props => props.borderSize} solid;
    border-radius: ${props => props.borderRadius};
    padding: 5px 15px;
    margin-right: 5px;
    &: hover {
        box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.8);
    }
`

const mapStateToProps = state => {
    return {
        bgColor: state.app.theme.colorContainerBg,
        borderSize: state.app.theme.borderSize,
        borderRadius: state.app.theme.borderRadius
    }
}

export default connect(mapStateToProps)(Button)
