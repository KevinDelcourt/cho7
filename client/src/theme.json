{
    "colors":{
        "connectionButton":"#916D43",
        "menuButton":"rgba(145, 109, 67, 0.35)",
        "submitButton":"#C4C4C4",
        "mainContainer": "rgba(145, 109, 67, 0.35)"
    },
    "border":{
        "mainContainerCorners":"20px",
        "mainContainer":"none"
    }

}