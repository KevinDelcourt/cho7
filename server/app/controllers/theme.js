const { queryCallback } = require("../../modules/queries")

const themeController = connection => {
    return {
        getTheme: (req, res) => {
            connection.query("SELECT * FROM theme", (err, rows) =>
                queryCallback(err, res, () => {
                    let theme = {}
                    rows.map(r => {
                        theme[r.style] = r.value
                    })
                    return res.send(theme)
                })
            )
        },

        updateTheme: (req, res) => {
            let query = {
                mysql: "INSERT INTO theme (style,value) VALUES ",
                data: []
            }

            const addValues = (key, value) => {
                query.mysql += "(?,?),"
                query.data.push(key)
                query.data.push(value)
            }

            for (let key in req.body)
                if (req.body[key] != "undefined") addValues(key, req.body[key])

            for (let key in req.files)
                addValues(key.slice(0, -4), req.files[key][0].originalname)

            query.mysql = query.mysql.slice(0, -1)
            connection.query("DELETE FROM theme", (err, rows) => {
                queryCallback(err, res, () => {
                    connection.query(query.mysql, query.data, (err, rows) => {
                        queryCallback(err, res, () => res.send(true))
                    })
                })
            })
        }
    }
}

module.exports = themeController
