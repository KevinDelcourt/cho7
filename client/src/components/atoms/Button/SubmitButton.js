import styled from "styled-components"
import { connect } from "react-redux"
import Button from "./Button"

const SubmitButton = styled(Button)`
    background-color: ${props => props.bgColor};
    border: ${props => props.borderSize} solid;
    border-radius: ${props => props.borderRadius};
`

const mapStateToProps = state => {
    return {
        bgColor: state.app.theme.colorSubmitBtn,
        borderSize: state.app.theme.borderSize,
        borderRadius: state.app.theme.borderRadius
    }
}

export default connect(mapStateToProps)(SubmitButton)
