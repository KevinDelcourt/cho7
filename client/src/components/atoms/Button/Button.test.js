import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Button from "./Button";
import theme from "../../../theme.json";

test("Button background-color renders correctly", () => {
  const tree = renderer
    .create(<Button bgColor={theme.color.brown1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("background-color", "rgba(145,109,67,0.35)");
});

test("Button color renders correctly", () => {
    const tree = renderer
      .create(<Button color={theme.color.white} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("color", "#fff");
  });

test("Button border renders correctly", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("border", "none");
});

//affichage deconnexion/connexion

//affichage, disparition boutons du menu

//modification du thème des boutons selon le créateur
