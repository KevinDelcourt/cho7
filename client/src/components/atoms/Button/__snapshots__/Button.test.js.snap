// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button background-color renders correctly 1`] = `
.c0 {
  background-color: rgba(145,109,67,0.35);
  border: none;
  border-radius: 5px;
  padding: 5px 15px;
  height: 35px;
  font-size: 18px;
  color: black;
}

<button
  className="c0"
/>
`;

exports[`Button border renders correctly 1`] = `
.c0 {
  border: none;
  border-radius: 5px;
  padding: 5px 15px;
  height: 35px;
  font-size: 18px;
  color: black;
}

<button
  className="c0"
/>
`;

exports[`Button color renders correctly 1`] = `
.c0 {
  border: none;
  border-radius: 5px;
  padding: 5px 15px;
  height: 35px;
  font-size: 18px;
  color: #fff;
}

<button
  className="c0"
  color="#fff"
/>
`;
